document/JS.bs

```diff
@@ -0,0 +1,921 @@
+    1. Let |memaddr| be |memory|.\[[Memory]].
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |store| be [=grow_mem=](|store|, |memaddr|, |delta|).
+    1. If |store| is [=error=], throw a [=RangeError=] exception.
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. Perform ! [=DetachArrayBuffer=](|memory|.\[[BufferObject]]).
+    1. Let |block| be a [=Data Block=] which is [=identified with=] the underlying memory of |memaddr|.
+    1. Let |buffer| be a new [=ArrayBuffer=] whose \[[ArrayBufferData]] is |block| and \[[ArrayBufferByteLength]] is set to the length of |m|.data.
+    1. Set |memory|.\[[BufferObject]] to |buffer|.
+    1. Return |ret|.
+</div>
+
+<h3 id="tables">Tables</h3>
+
+<pre class="idl">
+enum TableKind {
+  "anyfunc",
+};
+
+dictionary TableDescriptor {
+  required TableKind element;
+  required [EnforceRange] unsigned long initial;
+  [EnforceRange] unsigned long maximum;
+};
+
+[LegacyNamespace=WebAssembly, Constructor(TableDescriptor descriptor), Exposed=(Window,Worker,Worklet)]
+interface Table {
+  void grow([EnforceRange] unsigned long delta);
+  ExportedFunction? get([EnforceRange] unsigned long delta);
+  void set([EnforceRange] unsigned long delta, ExportedFunction? value);
+  readonly attribute unsigned long length;
+};
+</pre>
+
+<div>
+A {{Table}} object contains a single [=table instance=]
+which can be simultaneously referenced by multiple {{Instance}} objects. Each
+{{Table}} object has one internal slots:
+
+    * \[[Table]] : a [=table address=]
+    * \[[Values]] : a List whose elements are either null or {{ExportedFunction}}s.
+</div>
+
+<div algorithm>
+    To <dfn>create a table object</dfn> from a [=table instance=] |t|, perform the following steps:
+    1. Let |values| be a list whose length is that of |t|.elem where each element is null.
+    1. Return a new {{Table}} instance with \[[Table]] set to |t| and \[[Values]] set to |values|.
+</div>
+
+<div algorithm>
+    The <dfn constructor for="Table">Table(descriptor)</dfn> constructor, when invoked, performs the following steps:
+    1. Let |n| be |descriptor|.initial.
+    1. If |descriptor|.maximum is present, let |m| be |descriptor|.maximum; otherwise, let |m| be empty.
+    1. If |m| is not empty and |m| &lt; |n|, throw a {{RangeError}} exception.
+    1. Let |type| be the [=table type=] {𝗆𝗂𝗇 n, 𝗆𝖺𝗑 |m|} 𝖺𝗇𝗒𝖿𝗎𝗇𝖼.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let (|store|, |tableaddr|) be [=alloc_table=](|store|, |type|). <!-- TODO(littledan): Report allocation failure https://github.com/WebAssembly/spec/issues/584 -->
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. [=Create a table object=] from the table instance |tableaddr| and return the result.
+</div>
+
+<div algorithm>
+    The <dfn for="Table">grow(d)</dfn> method, when invoked, performs the following steps:
```

Comment:
This and others are missing `method` inside their `<dfn` open tag. You can tell because in the output they are not monospace.